### PR TITLE
Form state - One

### PR DESCRIPTION
**Pull request by cgcarlosg**
At this pull request, I add state to the Form component and add methods to create and remove books from the store

Uses useState hook to turn BooksForm stateful
Passes deleteBook method down as a prop from BookList to Book component
Creates handleRemoveBook to handle the event that deletes a book from the store
Adds input validation on BooksForm

### CODE REVIEW

File: src/containers/BookList.js [hunk 1]
```diff
@@ -2,16 +2,12 @@ import React from 'react';
...
-  <div>
-    {
-      books.map(
-        book => <Book book={book} key={book.id} delete={handleRemoveBook} />,
-      )
-    }
-  </div>
+const BookList = ({ books }) => (
+  books.map(
+    book => <tr key={book.id}><Book book={book} /></tr>,
```
Comment (afanimartin) [inline src/containers/BookList.js]: Please implement the `handleRemoveBook` function and pass it as a prop to the Book component.

File: src/components/Book.js [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
   <table className="table">
     <tr>
-      <td>{book.id}</td>
-      <td>{book.title}</td>
-      <td>{book.category}</td>
+      <td className="col-3">{book.id}</td>
+      <td className="col-3">{book.title}</td>
+      <td className="col-3">{book.category}</td>
+      <td className="col-3">
+        <button type="button" onClick={() => handleRemoveBook(book)}>
```
Comment (afanimartin) [inline src/components/Book.js:11]: Please implement the handRemoveBook function in the BookList component. This method should only be passed as a prop in this component.

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @cgcarlosg , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by eliasecasta — APPROVED** [review]
## Status:  Approved 🟢

Hi, @cgcarlosg, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**